### PR TITLE
PAYARA-737 fix GlobalTID.hashCode()

### PR DESCRIPTION
**Pull request by lprimak**
disregard branch qualifier in hashCode as pointed out by GLASSFISH-21532

### CODE REVIEW

**Comment by lprimak** [comment]
jenkins test please

File: appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/GlobalTID.java [hunk 1]
```diff
@@ -309,14 +308,16 @@ public final int hashCode() {
...
-        if( realTID.tid != null )
-            for( int pos = 0; pos < realTID.tid.length; pos++ )
+        // GLASSFISH-21532 The branch qualifier values should not be used for calculating the hashcode
+        if( realTID.tid != null ) {
+            for( int pos = 0; pos < realTID.tid.length - realTID.bqual_length; pos++ ) {
                 hashCode += realTID.tid[pos];
+            }
+        }
 
         // Add in the formatId and branch qualifier length.
```
Comment (Pandrex247) [inline appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/GlobalTID.java]: The most minor of complaints: can you fix this comment please? It no longer adds the bqual length

Comment (lprimak) [inline appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/GlobalTID.java]: Thx!

**Comment by lprimak** [comment]
jenkins test please

**Comment by lprimak** [comment]
jenkins test please

**Pull request merged**